extension/sigv4authextension/README.md

```diff
@@ -0,0 +1,51 @@
+# Authenticator - Sigv4
+
+This extension provides the Sigv4 process of adding authentication information to AWS API requests to HTTP based exporters for AWS services. This extension implements the `configauth.ClientAuthenticator` interface. For more information on the Sigv4 process, please look [here](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html).
+
+The authenticator type has to be set to `sigv4auth`. 
+
+## Configuration
+
+The configuration fields are as follows:
+
+* `region`: **Required**. The AWS region for AWS Sigv4
```

Comment:
Can't we infer region / service from URL?

Reply:
We instead try to infer region from the `AWS_DEFAULT_REGION` environment variable; we can infer region (and  service?) from a URL, and that's done through `parseEndpointRegion()` in the [AWS PRW Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awsprometheusremotewriteexporter/auth.go#L123), but [it requires a URL](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awsprometheusremotewriteexporter/auth.go#L123) from the [PRW configs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awsprometheusremotewriteexporter/auth.go#L123) when called which the extension will not have access to.

Reply:
The extension has access to the HTTP request so it should be able to read the URL out of it, let's infer from that.

Reply:
Would we want to look at both `AWS_REGION` and `AWS_DEFAULT_REGION`? If we looked at just one, `AWS_REGION` might make more sense since it seems to override `AWS_DEFAULT_REGION` ([AWS Env. Variables](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html))

Reply:
The URL won't be inferrable for service/region only for localhost development IIUC so having the fallback to config but not making them required parameters is still a good idea (global endpoints like iam.amazonaws.com won't have a region either way). We can probably skip checking those env variables since this will be used rarely only in AWS dev situations.

Reply:
@anuraaga added in the logic for inferring region from URL; currently, the order of precedence is: user configuration -> URL inference -> env variables. I've kept in the env variables check for now. Region is now an optional parameter.

Reply:
We should infer service too

Reply:
Let's go ahead and remove the environment variable support, we seem to be in a situation of worst of both worlds because we only support one form of env var but not all the ones the aws sdk supports. We could support both env vars, but it seems unnecessary if we can infer from the variable for non-aws-dev experiences, and expecting `region: ${AWS_REGION}` isn't the end of the world in the unlikely case that is actually needed.

Reply:
I think we should remove all non-explicit configuration.  I'm not 100% sold that the region inference from the URL is correct and even less sure of the idea of inferring the service.  We can always add features later, which I think would be preferable to adding potentially buggy features at the start.

Reply:
Hi @anuraaga, after discussing with @Aneurysm9, inferring region or service from URL seems to be a bit worrisome. For now, it might be better to leave out region/service inference and require explicit configuration, and it would be easier to add those features later than remove it later.

Reply:
Resolving conversation since region/service inference was added